client/components/account-balances/balances-tab-panel.tsx

```diff
@@ -74,100 +84,120 @@ const AccountBalancesTabPanel: React.FC = () => {
 		);
 	}
 
-	return (
-		<TabPanel tabs={ depositCurrencyTabs }>
-			{ ( tab: BalanceTab ) => (
-				<Flex gap={ 0 } className="wcpay-account-balances__balances">
-					<BalanceBlock
-						id={ `wcpay-account-balances-${ tab.currencyCode }-available` }
-						title={ fundLabelStrings.available }
-						amount={ tab.availableFunds }
-						currencyCode={ tab.currencyCode }
-						isLoading={ isLoading }
-						tooltip={
-							<BalanceTooltip
-								label={ `${ fundLabelStrings.available } tooltip` }
-								content={
-									tab.availableFunds < 0
-										? interpolateComponents( {
-												mixedString: __(
-													'{{learnMoreLink}}Learn more{{/learnMoreLink}} about why your account balance may be negative.',
-													'woocommerce-payments'
-												),
-												components: {
-													learnMoreLink: (
-														// eslint-disable-next-line jsx-a11y/anchor-has-content
-														<a
-															rel="external noopener noreferrer"
-															target="_blank"
-															href={
-																documentationUrls.negativeBalance
-															}
-														/>
+	// Selected currency is not valid if it is not in the list of deposit currencies.
+	const isSelectedCurrencyValid =
+		selectedCurrency &&
+		depositCurrencyTabs.some( ( tab ) => tab.name === selectedCurrency );
+
+	// Wrap TabPanel to allow a full re-render when the selected currency changes.
+	// since TabPanel will not allow controlled tab selection, we use the initialTabName prop.
+	const TabPanelComp: React.FC = () => {
+		return (
+			<TabPanel
+				tabs={ depositCurrencyTabs }
+				onSelect={ onTabSelect }
+				initialTabName={
+					isSelectedCurrencyValid ? selectedCurrency : undefined
+				}
+			>
+				{ ( tab: BalanceTab ) => (
```

Comment:
This diff looks extensive, however all code within `<TabPanel></TabPanel>` is a copy-paste from `develop`, with indentation changes.

Reply:
FYI there's a `Hide whitespace` option in the GitHub diff view, really useful for reviewing changes like this. 

In the gear menu: 

![Screenshot 2023-04-12 at 9 05 22 AM](https://user-images.githubusercontent.com/4167300/231288141-9044f026-a2ee-4a6b-97df-b3ae5a830a83.png)